torchgeo/samplers/batch.py

```diff
@@ -97,9 +98,11 @@ def __init__(
             roi: region of interest to sample from (minx, maxx, miny, maxy, mint, maxt)
                 (defaults to the bounds of ``dataset.index``)
             units: defines if ``size`` is in pixel or CRS units
+            generator: random number generator
```

Comment:
I would like to make our usage of `generator` consistent across all of TorchGeo. This includes the name, description, type, and default value. This isn't exactly consistent in our existing code, but I don't want to make it even more consistent. So let's decide on defaults for these and then make sure everything matches. Here is my suggestion:

* name: `generator`
* description: `Pseudo-random number generator (PRNG).`
* type: `Generator | None`
* default: `None`

If you agree with this, you or I can open a separate PR for existing code.

Reply:
Note that every function we add a new parameter to needs:
```rst
.. versionadded:: 0.7
   The *generator* parameter.
```

Reply:
> I would like to make our usage of `generator` consistent across all of TorchGeo. This includes the name, description, type, and default value. This isn't exactly consistent in our existing code, but I don't want to make it even more consistent. So let's decide on defaults for these and then make sure everything matches. Here is my suggestion:
> 
> * name: `generator`
> * description: `Pseudo-random number generator (PRNG).`
> * type: `Generator | None`
> * default: `None`
> 
> If you agree with this, you or I can open a separate PR for existing code.

My docstring was loosely based on Torch: "generator ([Generator](https://pytorch.org/docs/stable/generated/torch.Generator.html#torch.Generator)) – Generator used in sampling.", but I am totally fine with your proposed structure.